crates/core/transaction/src/plan/build.rs

```diff
@@ -188,273 +153,56 @@ impl TransactionPlan {
             memo,
         };
 
-        // TODO: add consistency checks?
-
-        Ok(UnauthTransaction {
-            inner: Transaction {
-                transaction_body,
-                anchor: witness_data.anchor,
-                binding_sig: [0; 64].into(),
-            },
-            synthetic_blinding_factor,
+        Ok(Transaction {
+            transaction_body,
+            anchor: witness_data.anchor,
+            binding_sig: [0; 64].into(),
         })
     }
 
-    #[cfg(feature = "parallel")]
-    /// Build the transaction this plan describes while proving concurrently.
-    /// This can be used in environments that support tokio tasks.
-    pub async fn build_concurrent<R: CryptoRng + RngCore>(
-        self,
-        rng: R,
-        fvk: &FullViewingKey,
-        witness_data: WitnessData,
-    ) -> Result<UnauthTransaction> {
-        let mut synthetic_blinding_factor = Fr::zero();
-
-        // Add the memo.
-        let mut memo: Option<MemoCiphertext> = None;
-        let mut memo_key: Option<PayloadKey> = None;
-        if self.memo_plan.is_some() {
-            let memo_plan = self
-                .memo_plan
-                .clone()
-                .ok_or_else(|| anyhow!("missing memo_plan in TransactionPlan"))?;
-            memo = memo_plan.memo().ok();
-            memo_key = Some(memo_plan.key);
+    /// Slot in the [`AuthorizationData`] and derive the synthetic
+    /// blinding factors needed to compute the binding signature
+    /// and assemble the transaction.
+    /// Arguments:
+    ///     self: `TransactionPlan`
+    ///     rng: `&mut R`
+    ///     auth_data: `&AuthorizationData`
+    ///     transaction: `Transaction`
+    /// Returns: `Transaction`
+    pub fn authorize_with_auth<R: CryptoRng + RngCore + Debug>(
+        &self,
+        rng: &mut R,
+        auth_data: &AuthorizationData,
+        mut transaction: Transaction,
+    ) -> Result<Transaction> {
```

Comment:
Rather than `authorize_with_auth`, could this just be `authorize`?

In terms of API design, the reason that I suggested putting this on the `TransactionPlan`, rather than the `Transaction`, is so that the method is only relevant when you're already building transactions. Because of that, I think it's fine to just call it `authorize`.